File: client-frontend/src/services/order.ts
import api from './api';
import type { ShopAddress } from './auth';

export interface Order {
  id: number;
  orderNo: string;
  status: string;
  source: string | null;
  startTime: string;
  endTime: string;
  serviceType: string | null;
  remark: string | null;
  address: string | null;
  quotePrice: number | null;
  quoteRemark: string | null;
  quotedAt: string | null;
  depositAmount: number;
  isDepositPaid: boolean;
  technician: {
    id: number;
    name: string;
    phone: string;
  } | null;
  customer: {
    id: number;
    name: string;
    phone: string;
  } | null;
  clientAddress: {
    doorInfo?: string;
    detailAddress?: string;
    province?: string;
    city?: string;
    district?: string;
  } | null;
  createdAt: string;
  updatedAt: string;
}

export interface CreateOrderDto {
  serviceDate: string;
  startTime: string;
  addressId?: number;
  techId: number;
  serviceType: string;
  selectedServiceIds?: string[];
  remark?: string;
  shopAddress?: ShopAddress;
}

export interface UpdateOrderDto {
  serviceDate: string;
  startTime: string;
  addressId: number;
}

export interface UpdateOrderStatusDto {
  status: 'completed' | 'cancelled';
}

export interface CreateOrderFromDesignDto {
  designId: number;
  techId: number;
  serviceDate: string;
  startTime: string;
  serviceType: string;
  addressId?: number;
  shopAddress?: {
    name: string;
    phone?: string;
    province?: string;
    city?: string;
    district?: string;
    detailAddress?: string;
    doorInfo?: string;
  };
}

export const orderService = {
  async getOrders(): Promise<Order[]> {
    const response = await api.get('/orders');
    return response.data;
  },

  async getTrips(): Promise<Order[]> {
    const response = await api.get('/orders/trips');
    return response.data;
  },

  async getOrder(id: number): Promise<Order> {
    const response = await api.get(`/orders/${id}`);
    return response.data;
  },

  async createOrder(data: CreateOrderDto): Promise<Order> {
    const response = await api.post('/orders', data);
    return response.data;
  },

  async createOrderFromDesign(data: CreateOrderFromDesignDto): Promise<Order> {
    const response = await api.post('/orders/from-design', data);
    return response.data;
  },

  async updateOrder(id: number, data: UpdateOrderDto): Promise<Order> {
    const response = await api.patch(`/orders/${id}`, data);
    return response.data;
  },

  async agreeQuote(id: number): Promise<Order> {
    const response = await api.post(`/orders/${id}/agree`, {});
    return response.data;
  },

  async rejectQuote(id: number, reason: string): Promise<Order> {
    const response = await api.post(`/orders/${id}/reject-quote`, { reason });
    return response.data;
  },

  async updateOrderStatus(id: number, data: UpdateOrderStatusDto): Promise<Order> {
    const response = await api.patch(`/orders/${id}/status`, data);
    return response.data;
  },
};

File: ios/NailBook/Models/Order.swift
import Foundation

// MARK: - Order Models

struct Order: Codable, Identifiable {
    let id: Int
    let orderNo: String
    var status: String
    var serviceType: String?
    var quotePrice: Double?
    var startTime: String?
    var endTime: String?
    var address: String?
    var remark: String?
    var customTitle: String?
    var customDescription: String?
    var customImages: [String]?
    var clientPhotos: [String]?
    var isDepositPaid: Bool?
    var depositAmount: Double?
    var quoteRemark: String?
    var cancelReason: String?
    var confirmedAt: String?
    var completedAt: String?
    var cancelledAt: String?
    var createdAt: String?
    var technician: OrderTechnician?
    var customer: OrderCustomer?
    var clientUser: OrderClientUser?
    var addressDetail: OrderAddress?
}

struct OrderTechnician: Codable, Identifiable {
    let id: Int
    var name: String?
    var avatarUrl: String?
    var phone: String?
}

struct OrderCustomer: Codable, Identifiable {
    let id: Int
    var name: String?
    var phone: String?
}

struct OrderClientUser: Codable, Identifiable {
    let id: Int
    var nickname: String?
    var avatarUrl: String?
}

struct OrderAddress: Codable, Identifiable {
    let id: Int
    var contactName: String?
    var contactPhone: String?
    var province: String?
    var city: String?
    var district: String?
    var detailAddress: String?
}

// MARK: - Order Status

enum OrderStatus: String, Codable, CaseIterable {
    case pendingQuote = "pending_quote"
    case quoted
    case pendingConfirm = "pending_confirm"
    case pendingHome = "pending_home"
    case pendingShop = "pending_shop"
    case inProgress = "in_progress"
    case completed
    case cancelled
    case expired

    var displayName: String {
        switch self {
        case .pendingQuote: return "待报价"
        case .quoted: return "已报价"
        case .pendingConfirm: return "待确认"
        case .pendingHome: return "待上门"
        case .pendingShop: return "待到店"
        case .inProgress: return "服务中"
        case .completed: return "已完成"
        case .cancelled: return "已取消"
        case .expired: return "已过期"
        }
    }
}
